### PR TITLE
fix: do not run generator without hilla

### PR DESCRIPTION
**Pull request by caalador**
Do not add the generators
if not hilla used in project.

Fixes vaadin/platform#5016

### CODE REVIEW

File: flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeTasks.java [hunk 1]
```diff
@@ -315,7 +315,9 @@ private void addEndpointServicesTasks(Options options) {
         EndpointGeneratorTaskFactory endpointGeneratorTaskFactory = lookup
                 .lookup(EndpointGeneratorTaskFactory.class);
 
-        if (endpointGeneratorTaskFactory != null) {
+        if (endpointGeneratorTaskFactory != null && FrontendUtils
```
Comment (mcollovati) [inline flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeTasks.java:318]: Could the whole `addEndpointServicesTasks` method be skipped if Hilla is not used?

Comment (caalador) [reply flow-server/src/main/java/com/vaadin/flow/server/frontend/NodeTasks.java]: Well there is a comment that it is skipped inside the method. Moved the isUsed check to be first.

File: flow-plugins/flow-plugin-base/src/test/java/com/vaadin/flow/plugin/base/BuildFrontendUtilTest.java [hunk 2]
```diff
@@ -95,7 +96,9 @@ public void should_notUseHilla_inPrepareFrontend()
 
         BuildFrontendUtil.prepareFrontend(adapter);
 
-        Mockito.verify(lookup).lookup(EndpointGeneratorTaskFactory.class);
+        // no hilla no lookup call
+        Mockito.verify(lookup, Mockito.never())
+                lookup(EndpointGeneratorTaskFactory.class);
```
Comment (mcollovati) [inline flow-plugins/flow-plugin-base/src/test/java/com/vaadin/flow/plugin/base/BuildFrontendUtilTest.java:101]:
```suggestion
                .lookup(EndpointGeneratorTaskFactory.class);
```

**Review by mcollovati — CHANGES_REQUESTED** [review]
There's a compilation error in one test, but other than that LGTM

File: flow-plugins/flow-plugin-base/src/test/java/com/vaadin/flow/plugin/base/BuildFrontendUtilTest.java (hunk 2 above)
Comment (caalador) [reply flow-plugins/flow-plugin-base/src/test/java/com/vaadin/flow/plugin/base/BuildFrontendUtilTest.java]: Done.

**Review by mcollovati — APPROVED** [review]

**Pull request merged**